communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,246 @@
+        return map.computeIfAbsent(gav, (k) -> new GAVDependencyTree(k));
+    }
+
+    private GAV generateGAV(ProjectVersionRef ref) {
+        return new GAV(ref.getGroupId(), ref.getArtifactId(), ref.getVersionString());
+    }
+
+    private DiscoveryResult getDiscoveryResult(File tempDir, File repoDir, ProjectVersionRef ref)
+            throws GalleyMavenException, URISyntaxException, CartoDataException {
+
+        MavenPomReader mavenPomReader = carto.getGalley().getPomReader();
+        MavenPomView pomView = mavenPomReader.read(ref, getRepoLocations(tempDir, repoDir));
+
+        Location location = new SimpleLocation("file:" + tempDir.getAbsolutePath());
+        URI src = new URI(location.getUri());
+
+        DiscoveryConfig disConf = new DiscoveryConfig(src);
+        disConf.setIncludeBuildSection(false);
+        disConf.setIncludeManagedDependencies(false);
+        disConf.setIncludeManagedPlugins(false);
+
+        MavenModelProcessor processor = new MavenModelProcessor();
+        return processor.readRelationships(pomView, src, disConf);
+    }
+
+    private List<Location> getRepoLocations(File cacheDir, File repoDir) {
+
+        Location location = new SimpleLocation("file:" + cacheDir.getAbsolutePath());
+
+        List<Location> repoLocations = new LinkedList<>();
+        // add local location
+        repoLocations.add(location);
+
+        // add maven central location
+        repoLocations.add(new SimpleLocation("central", "http://repo.maven.apache.org/maven2/"));
+
+        // add all the repositories mentioned in the pom.xml
+        List<SimpleLocation> locationsInPom = getLocationsDefinedInPom(repoDir);
+        repoLocations.addAll(locationsInPom);
+
+        return repoLocations;
+    }
+
+    private List<SimpleLocation> getLocationsDefinedInPom(File repoDir) {
+
+        List<String> pomFilePaths = findAllPomFiles(repoDir);
+        List<SimpleLocation> locations = new LinkedList<>();
+
+        for (String pom : pomFilePaths) {
+            Optional<MavenProject> project = pomReader.analyze(new File(pom));
+            if (project.isPresent() && project.get().getMavenRepositories() != null) {
+                for (MavenRepository repo : project.get().getMavenRepositories()) {
+                    locations.add(new SimpleLocation(repo.getId(), repo.getUrl()));
+                }
+            }
+        }
+
+        return locations;
+    }
+
+    /**
+     * Given the locations of the pom.xml files, setup the tempDir into a structure
+     * where Cartographer will be able to analyze the poms, and return back a
+     * list of ProjectVersionRefs objects generated from the pom.xml files.
+     * @param tempDir
+     * @param poms
+     * @return
+     * @throws TransferException
+     * @throws IOException
+     */
+    private Map<File, ProjectVersionRef> getProjectVersionRefs(File tempDir, List<String> poms)
+            throws TransferException, IOException {
```

Comment:
Do I get it correctly? This method populates temDir with maven repo like directory structure and put pom files to appropriate directories?